src/lib/client.js

```diff
@@ -3,23 +3,27 @@
 const assert = require('assert')
 const WebSocket = require('ws')
 const debug = require('debug')('ilp-plugin-payment-channel-framework:client')
-const url = require('url')
+const { URL } = require('url')
 
 module.exports = class BtpClient {
-  constructor ({ server, secret, plugin, insecure }) {
-    this._server = server
+  constructor ({ btpUri, plugin }) {
     this._plugin = plugin
 
-    // The server URI must follow the format: btp+wss://host:port/path
-    // See also: https://github.com/interledger/interledger/wiki/Interledger-over-BTP
-    const parsedServer = url.parse(server)
-    assert(parsedServer.protocol.startsWith('btp+'), 'server uri must start with "btp+"')
-    this._wsUri = (insecure ? 'ws://' : 'wss://') + parsedServer.host + parsedServer.path
-    this._secret = (parsedServer.auth && parsedServer.auth.split(':')[1]) || secret
-
-    if (!secret) {
-      throw new Error('secret must be provided to BTP Client in the URL or in the configuration')
-    }
+    // The BTP URI must follow one of the following formats:
```

Comment:
another supported format is `btp+wss://:auth_token@host/path` or `btp+wss://host/path`

Reply:
the first one is just `auth_username === ''`, that's fine. The second one is not supported I think (but we should discuss that in https://github.com/interledger/rfcs/pull/300)